refinery/ui/source/js/data-set-import/controllers/file-upload-s3.js

```diff
@@ -5,25 +5,133 @@
     .module('refineryDataSetImport')
     .controller('RefineryFileUploadS3Ctrl', RefineryFileUploadS3Ctrl);
 
-  RefineryFileUploadS3Ctrl.$inject = ['s3UploadService'];
+  RefineryFileUploadS3Ctrl.$inject = ['s3UploadService', '$log', '$scope'];
 
-  function RefineryFileUploadS3Ctrl (s3UploadService) {
+  function RefineryFileUploadS3Ctrl (s3UploadService, $log, $scope) {
     var vm = this;
-    vm.uploadFiles = function (files) {
-      vm.files = files;
-      if (files && files.length > 0) {
-        angular.forEach(vm.files, function (file) {
-          s3UploadService.upload(file).then(function () {
-            // Mark as success
-            file.success = true;
-          }, function (error) {
-            // Mark the error
-            vm.error = error;
-          }, function (progress) {
-            // Write the progress as a percentage
+    vm.multifileUploadInProgress = false;
+
+    vm.isUploadConfigured = function () {
+      if (s3UploadService.isConfigReady) {
+        return s3UploadService.isConfigValid();
+      }
+      return false;
+    };
+
+    vm.addFiles = function (files) {
+      if (typeof vm.files === 'undefined') {
+        vm.files = files;
+      } else {
+        vm.files = vm.files.concat(files);
+      }
+    };
+
+    vm.isFileNew = function (file) {
+      // check if file upload has been attempted
+      return typeof file.progress === 'undefined';
+    };
+
+    vm.isUploadInProgress = function (file) {
+      if (vm.isFileNew(file) || file.$error || file.success) {
+        return false;
+      }
+      return file.progress <= 100;
+    };
+
+    vm.areUploadsInProgress = function () {
+      if (typeof vm.files === 'undefined') {
+        return false;
+      }
+      return vm.files.some(vm.isUploadInProgress);
+    };
+
+    vm.isUploadComplete = function (file) {
+      return Boolean(file.$error || file.success);
+    };
+
+    vm.areUploadsCancellable = function () {
+      if (typeof vm.files === 'undefined') {
+        return false;
+      }
+      return !vm.files.every(vm.isUploadComplete);
+    };
+
+    vm.areUploadsEnabled = function () {
+      if (typeof vm.files === 'undefined' || vm.areUploadsInProgress()) {
+        return false;
+      }
+      return vm.files.some(vm.isFileNew);
+    };
+
+    vm.uploadFile = function (file) {
+      try {
+        file.managedUpload = s3UploadService.upload(file);
+      } catch (e) {
+        file.progress = 100;
+        file.$error = 'Data upload configuration error';
```

Comment:
It is already logged to console in `s3UploadService` and I'll another log statement there too.